src/test/run-pass-fulldeps/derive-no-std.rs

```diff
@@ -8,12 +8,13 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-#![feature(rand, collections, rustc_private)]
+#![feature(libc, lang_items, start)]
 #![no_std]
 
-extern crate rand;
-extern crate serialize as rustc_serialize;
-extern crate collections;
+extern crate libc; // to pull in libSystem on OSX
+
+#[lang = "panic_fmt"] fn panic_fmt() -> ! { loop {} }
+#[lang = "eh_personality"] extern fn eh_personality() {}
```

Comment:
Ah OK. So then the test binary itself will use `std`? In that case we can use `assert!` and, ya know, actually test stuff.